client/service-container-chain/src/cli.rs

```diff
@@ -45,16 +48,82 @@ pub struct ContainerChainRunCmd {
     /// Keep container-chain db after changing collator assignments
     #[arg(long)]
     pub keep_db: bool,
+
+    /// Creates a less resource-hungry node that retrieves relay chain data from an RPC endpoint.
+    ///
+    /// The provided URLs should point to RPC endpoints of the relay chain.
+    /// This node connects to the remote nodes following the order they were specified in. If the
+    /// connection fails, it attempts to connect to the next endpoint in the list.
+    ///
+    /// Note: This option doesn't stop the node from connecting to the relay chain network but
+    /// reduces bandwidth use.
+    #[arg(
+		long,
+		value_parser = validate_relay_chain_url,
+		num_args = 0..,
+		alias = "relay-chain-rpc-url"
+    )]
+    pub relay_chain_rpc_urls: Vec<Url>,
+
+    /// EXPERIMENTAL: Embed a light client for the relay chain. Only supported for full-nodes.
+    /// Will use the specified relay chain chainspec.
+    #[arg(long, conflicts_with_all = ["relay_chain_rpc_urls", "collator"])]
+    pub relay_chain_light_client: bool,
+}
+
+impl ContainerChainRunCmd {
+    /// Create a [`NormalizedRunCmd`] which merges the `collator` cli argument into `validator` to
+    /// have only one.
+    pub fn normalize(&self) -> ContainerChainCli {
+        let mut new_base = self.clone();
+
+        new_base.base.validator = self.base.validator || self.collator;
+
+        // Append `containers/` to base_path for this object. This is to ensure that when spawning
+        // a new container chain, its database is always inside the `containers` folder.
+        // So if the user passes `--base-path /tmp/node`, we want the ephemeral container data in
+        // `/tmp/node/containers`, and the persistent storage in `/tmp/node/config`.
+        // TODO: there should be a way to avoid this if we refactor the code that creates the db,
+        // but maybe that breaks dancebox
+        let base_path = match self.base.base_path() {
+            Ok(Some(x)) => x,
+            _ => {
+                // This is maybe unreachable. There is always a default base path, and if run in
+                // `--dev` or `--tmp` mode, a temporary base path is created.
+                panic!("No base path")
```

Comment:
This panic is not unreachable, it can be reached by `./target/release/tanssi-node solo-chain`

Reply:
*fixed*